pkg/apis/core/validation/utils.go

```diff
@@ -122,11 +122,17 @@ func shootReconciliationSuccessful(shoot *core.Shoot) (bool, string) {
 		return false, "no conditions and last operation present yet"
 	}
 
-	if shoot.Status.LastOperation != nil &&
-		shoot.Status.LastOperation.Type == core.LastOperationTypeReconcile &&
-		shoot.Status.LastOperation.State == core.LastOperationStateSucceeded {
-		return true, ""
+	if shoot.Status.LastOperation != nil {
+		if shoot.Status.LastOperation.Type == core.LastOperationTypeCreate ||
+			shoot.Status.LastOperation.Type == core.LastOperationTypeReconcile ||
+			shoot.Status.LastOperation.Type == core.LastOperationTypeRestore {
+			if shoot.Status.LastOperation.State != core.LastOperationStateSucceeded {
+				return false, "last operation type was create, reconcile or restore but state was not succeeded"
+			}
+		} else if shoot.Status.LastOperation.Type == core.LastOperationTypeMigrate {
+			return false, "last operation type was migrate, the migration process is not finished yet"
```

Comment:
Hm, updating to a new Kubernetes version would still work here, or? Why do you exclude this case?

Reply:
I think we should allow only Succeeded states of Reconcile and Creat. The migration flow doesn't reconcile all the components.

Reply:
OK, fair enough